mockall_derive/src/mock_function.rs

```diff
@@ -472,7 +472,23 @@ impl MockFunction {
             };
             (&self.call_vis, dead_code)
         };
-        let substruct_obj = if let Some(trait_) = &self.trait_ {
+        // Add #[no_mangle] attribute to preserve the function name
+        // as-is, without mangling, for compatibility with C functions.
+        let no_mangle = if let Some(ref abi) = self.sig.abi {
+            if let Some(ref name) = abi.name {
+                if name.value().ne("Rust") {
+                    quote!(#[allow(no_mangle)])
```

Comment:
What is "allow" for?  I thought that was just for quieting lints.  I think the correct attribute is
```suggestion
                    quote!(#[no_mangle])
```